libs/common/src/vault/models/api/cipher-permissions.api.ts

```diff
@@ -24,7 +24,9 @@ export class CipherPermissionsApi extends BaseResponse implements SdkCipherPermi
   /**
    * Converts the SDK CipherPermissionsApi to a CipherPermissionsApi.
    */
-  static fromSdkCipherPermissions(obj: SdkCipherPermissions): CipherPermissionsApi | undefined {
+  static fromSdkCipherPermissions(
+    obj: SdkCipherPermissions | undefined,
+  ): CipherPermissionsApi | undefined {
```

Comment:
:pick: Making the `obj` optional provides more flexibility when invoking the method. `| undefined` requires that a value be passed where as making the `obj` optional allows no parameters.

[TSPlayground](https://www.typescriptlang.org/play/?ssl=6&ssc=26&pln=1&pc=1#code/MYewdgzgLgBAZgVzMAjDAvDAFCARgKwC4ZoAnASzAHMYAfGJAEwFM5LnGBKDAPhj3wBuAFChIsRMgBMGbAID8xMpSrd0fASOGTUWbgHp9MSgDcAhgBtyjGAAMoACxVYmrdl1vGIMMMw4dtJGApPRhDGHMrRkDpFzAWNl8uMKNI6yA)
```ts
const func1 = (obj: string | undefined) => obj;
const func2 = (obj?: string) => obj;

func1() // invalid `thing(undefined)` is needed
func2() // valid
func2(undefined) // valid
```

```suggestion
    static fromSdkCipherPermissions(obj?: SdkCipherPermissions): CipherPermissionsApi | undefined {
```

Reply:
You have similar patterns in other domain classes, I won't spam the same comment on each one. I can live with either but we should be consistent whichever way you choose to go!